src/main/java/com/iota/iri/service/tipselection/impl/RecursiveWeightCalculator.java

```diff
@@ -75,21 +78,36 @@ public RecursiveWeightCalculator(Tangle tangle, SnapshotProvider snapshotProvide
         return hashWeight;
     }
 
-    private int getRating(HashSet<Hash> nonDupes, Map<Hash, HashSet<Hash>> txToDirectApprovers) throws Exception {
-        Deque<Hash> stack = new ArrayDeque<>(nonDupes);
-        while (CollectionUtils.isNotEmpty(stack)) {
-            HashSet<Hash> approvers = getTxDirectApproversHashes(stack.pollLast(), txToDirectApprovers);
+    /**
+     * Gets the rating of a set, calculated by checking its approvers
+     * 
+     * @param startingSet
+     * @param txToDirectApproversCache 
+     * @return
+     * @throws Exception
+     */
+    private int getRating(Set<Hash> startingSet, Map<Hash, Set<Hash>> txToDirectApproversCache) throws Exception {
```

Comment:
Change the name to `getWeight`.

Optional:
I think it will be cleaner to add + 1 when you return at the end instead. Or conversely instead of `startingSet` you can have `startingHash` or `entryHash`. But then you will have that `getApprovers` call that you did before in `calculateRatingDfs`. It is presumably not so bad because you have the cache and the code will become more readable (you get the weight of the tx in the param).